src/mkMarkdown.ts

```diff
@@ -0,0 +1,25 @@
+import * as vscode from 'vscode';
+import { splitIfPanelExists } from './extension';
+
+export function mkMarkdown(): (...args: any[]) => any {
+  return async () => {
+    const currentPanel = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+    const uri = vscode.Uri.parse('markdown:' + 'L4'); // 'name of tab itself 
+    const doc = await vscode.workspace.openTextDocument(uri); // calls back into the provider
+
+    // ensure new panel opens instead of new tab
+    splitIfPanelExists(currentPanel);
+
+    await vscode.window.showTextDocument(doc, vscode.ViewColumn.Beside);
+  };
+}
+
+ // Register virtual doc provider
+ export const myScheme = 'markdown';
+ export const myProvider = new class implements vscode.TextDocumentContentProvider {
```

Comment:
`myScheme` and `myProvider` are a bit vague. Not easy to see what they are.

Reply:
I've edited it so it reads like so: registerTextDocumentContentProvider('markdown', markdownProvider)